pymeasure/display/widgets/log_widget.py

```diff
@@ -25,38 +25,148 @@
 import logging
 
 from ..log import LogHandler
-from ..Qt import QtWidgets
+from ..Qt import QtWidgets, QtCore, QtGui
 from .tab_widget import TabWidget
 
 log = logging.getLogger(__name__)
 log.addHandler(logging.NullHandler())
 
 
+class HTMLFormatter(logging.Formatter):
+    level_colors = {
+        "DEBUG": "DarkGray",
+        # "INFO": "Black",
+        "WARNING": "DarkOrange",
+        "ERROR": "Red",
+        "CRITICAL": "DarkRed",
+    }
+
+    html_replacements = {
+        "\r\n": "<br>  ",
+        "\n": "<br>  ",
+        "\r": "<br>  ",
+        "  ": "&nbsp;" * 2,
+        "\t": "&nbsp;" * 4,
+    }
+
+    def format(self, record):
+        formatted = super().format(record)
+
+        # Apply color if a level-color is defined
+        if record.levelname in self.level_colors:
+            formatted = f"<font color=\"{self.level_colors[record.levelname]}\">{formatted}</font>"
+
+        # ensure newlines and indents are preserved
+        for replacement in self.html_replacements.items():
+            formatted = formatted.replace(*replacement)
+
+        # Prepend the level as HTML comment
+        formatted = f"<!--{record.levelname}-->{formatted}"
+
+        return formatted
+
+
 class LogWidget(TabWidget, QtWidgets.QWidget):
     """ Widget to display logging information in GUI
 
     It is recommended to include this widget in all subclasses of
     :class:`ManagedWindowBase<pymeasure.display.windows.managed_window.ManagedWindowBase>`
     """
 
-    def __init__(self, name, parent=None):
+    fmt = '%(asctime)s : %(message)s (%(levelname)s)'
+    datefmt = '%m/%d/%Y %I:%M:%S %p'
+
+    tab_widget = None
+    tab_index = None
+
+    blink_qtimer = QtCore.QTimer()
+    blink_color = None
+    blink_state = False
+
+    def __init__(self, name, parent=None, fmt=None, datefmt=None):
+        if fmt is not None:
+            self.fmt = fmt
+        if datefmt is not None:
+            self.datefmt = datefmt
```

Comment:
Regarding the default values: Why don't you put the defaults into the signature?
That way everybody would know, what the defaults are and could orient at that value:
```suggestion
    tab_widget = None
    tab_index = None

    blink_qtimer = QtCore.QTimer()
    blink_color = None
    blink_state = False

    def __init__(self, name, parent=None,
                 fmt='%(asctime)s : %(message)s (%(levelname)s)',
                 datefmt='%m/%d/%Y %I:%M:%S %p',
                 ):
        self.fmt = fmt
        self.datefmt = datefmt
```

Reply:
Ah, I see the reason know: Below, you want to pass "None" to the constructor, unless the user changed it.

Reply:
Indeed, I wanted to put them as default values in there indeed, but then we need to either place them as default in all of the valling classes, or use some kwargs dictionary, but that seemed to result in relativity much needless code..

Reply:
Is that okay for you like this? We do have other cases where the default is manually copied into the calling classes (e.g. linewidth for the plot widget), so we could also do that, but this is a relatively long default value (I find it a bit of an ugly thing that there is no nice way to do this...)

Reply:
@bmoneke, is it ok for you to merge?

Reply:
Yes it is, as there is no perfect solution.